x/stakeibc/keeper/community_pool_test.go

```diff
@@ -166,3 +169,116 @@ func (s *KeeperTestSuite) TestQueryCommunityPoolBalance_FailedQuerySubmission()
 	err := s.App.StakeibcKeeper.QueryCommunityPoolBalance(s.Ctx, invalidHostZone, icaAccountType, Atom)
 	s.Require().ErrorContains(err, "Error submitting query for pool ica balance")
 }
+
+
+
+type RedeemCommunityPoolTokensTestCase struct {
+	hostZone                     types.HostZone
+	initialDummyTokens           sdkmath.Int
+	communityPoolHoldingAddress  sdk.AccAddress
+}
+
+func (s *KeeperTestSuite) SetupRedeemCommunityPoolTokens() RedeemCommunityPoolTokensTestCase {
+	s.CreateTransferChannel(HostChainId)
+
+	// Create relevant module and ICA accounts
+	depositAddress := s.TestAccs[0]
+	communityPoolHoldingAddress := s.TestAccs[1]
+	communityPoolReturnICAAddress := DelegationICAAddress // need an address on HostChain (starts cosmos)
+
+	// native stTokens which will be redeemed, dummy tokens which should not be touched
+	initialNativeStTokens := sdk.NewInt(1000)
+	initialDummyTokens := sdk.NewInt(999)
+
+	// Fund the redeem holding address with native staked tokens and
+	// some dummy tokens that should not get touched while redeeming
+	stDenom := types.StAssetDenomFromHostZoneDenom(Atom)
+	s.FundAccount(communityPoolHoldingAddress, sdk.NewCoin(stDenom, initialNativeStTokens))
+	s.FundAccount(communityPoolHoldingAddress, sdk.NewCoin(Atom, initialDummyTokens))   // dummy token
+	s.FundAccount(communityPoolHoldingAddress, sdk.NewCoin(IbcAtom, initialDummyTokens)) // dummy token
+
+	// Create a host zone with valid addresses to perform the liquid stake
+	hostZone := types.HostZone{
+		ChainId:                           HostChainId,
+		Bech32Prefix:                      Bech32Prefix,
+		HostDenom:                         Atom,
+		IbcDenom:                          IbcAtom,
+		TransferChannelId:                 ibctesting.FirstChannelID,
+		CommunityPoolRedeemHoldingAddress: communityPoolHoldingAddress.String(),
+		CommunityPoolReturnIcaAddress:     communityPoolReturnICAAddress,
+		DepositAddress:                    depositAddress.String(),
+		TotalDelegations:                  initialNativeStTokens, // at least as much as we are trying to redeem
+		RedemptionRate:                    sdk.OneDec(),
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	// Create the epoch tracker and deposit records so the liquid stake succeeds
+	epochNumber := uint64(1)
+	epochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.DAY_EPOCH,
+		EpochNumber:        epochNumber,
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 30_000_000), // dictates transfer timeout
```

Comment:
Double check me, but I think we can remove the next epoch start time cause it shouldnt be used by this function